src/test/java/htsjdk/samtools/reference/FastaReferenceWriterTest.java

```diff
@@ -195,6 +192,18 @@ public void testAddingInvalidSequenceName(final String invalidName) throws IOExc
         }
     }
 
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void testAddingGziIndexToNonBCFFile() throws IOException {
+        final Path testOutput = File.createTempFile("fwr-test", ".fasta").toPath();
+        final Path testOutputGZI = File.createTempFile("fwr-test", ".fasta.gzi").toPath();
+        IOUtil.deleteOnExit(testOutput);
+        Files.delete(testOutput);
```

Comment:
Not sure these files need to actually exist for this test, but they only need to be deleted once. If you're trying to get the index deleted as well, I think you need to create a temp dir (I don't think there is any guarantee that two temp files are siblings) ?

Reply:
I set both files to delete on exit to be sure. It doesn't matter for this test since this is just asserting validation behavior on the `.build()` step.